src/queries/__tests__/role.test.tsx

```diff
@@ -134,4 +134,43 @@ describe('supports name option', () => {
       'target-button'
     );
   });
+
+  test('returns an element when the direct child is text', () => {
+    const { getByRole } = render(
+      <Text accessibilityRole="header" testID="target-header">
+        About
+      </Text>
+    );
+
+    // assert on the testId to be sure that the returned element is the one with the accessibilityRole
+    expect(getByRole('header', { name: 'About' }).props.testID).toBe(
+      'target-header'
+    );
```

Comment:
Using `getByTestId` to fetch the host element seems to be a bit stronger assumption on getRoleBehavior. But it makes assumption about `getByTestId` being correctly implemented.
```suggestion
    expect(getByRole('header', { name: 'About' })).toBe(getByTestId('target-header'));
    expect(getByRole('header', { name: 'About' }).props.testID).toBe(
      'target-header'
    );
```